### PR TITLE
Adding IUCN RedList category to the species match cache

### PR DESCRIPTION
**Pull request by fmendezh**
https://github.com/gbif/pipelines/issues/257

### CODE REVIEW

File: kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java [hunk 1]
```diff
@@ -105,34 +107,47 @@ private static void run(NameUsageMatchIndexingOptions options) {
...
                         NameUsageMatchKVStoreFactory.valueMutator(
                             Bytes.toBytes(storeConfiguration.getHBaseKVStoreConfiguration().getColumnFamily()),
                             Bytes.toBytes(storeConfiguration.getValueColumnQualifier()));
                   }
 
+                  /**
+                   * Retrieves the IUCN Red List category from the name usage match response.
+                   */
+                  private ThreatStatus getIucnRedListCategory(NameUsageMatch nameUsageMatch) {
+                    Integer usageKey = Objects.nonNull(nameUsageMatch.getAcceptedUsage()) ? nameUsageMatch.getAcceptedUsage().getKey() : nameUsageMatch.getUsage().getKey();
```
Comment (marcos-lg) [inline kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java:127]: I'm not 100% sure but I think there could be cases where nameUsageMatch is not null but both the acceptedUsage and the usage are null. Maybe it's worth to handle that case

File: kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java [hunk 2]
```diff
@@ -105,34 +107,47 @@ private static void run(NameUsageMatchIndexingOptions options) {
...
+                    Map<String,String> category = checklistbankService.getIucnRedListCategory(usageKey);
+                    if (Objects.nonNull(category)) {
+                      return ThreatStatus.valueOf(category.get("category"));
```
Comment (marcos-lg) [inline kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java:130]: Same as above, thinking if we should handle cases where the category is not present in the enum

**Review by marcos-lg — COMMENTED** [review]
Just those comments about checking unexpected responses and if we should set the iucn field to null (or something else) or let it break and skip those records

File: kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java (hunk 1 above)
Comment (fmendezh) [reply kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java:127]: ok I will double check those nulls then

Comment (fmendezh) [reply kvs-indexing/src/main/java/org/gbif/kvs/indexing/species/NameUsageMatchIndexer.java]: https://github.com/gbif/key-value-store/commit/abb666a9a6b28f0fb8209f9f62115398c4188d21

**Pull request merged**